src/restic/node.go

```diff
@@ -581,6 +581,7 @@ func (node *Node) fillExtra(path string, fi os.FileInfo) error {
 	case "file":
 		node.Size = uint64(stat.size())
 		node.Links = uint64(stat.nlink())
+		node.Device = uint64(stat.dev())
```

Comment:
I am afraid we're mixing up things here. Device is now used twice. Both as st_dev and as st_rdev. Which means if I make a hardlinked /dev/null, restic will store 518 (1,3) in the Device field and during restore this will never match any of the stored inodes in the HardlinkIndex, right?